### ISSUE TITLE
Add Lists API

### ISSUE THREAD

**Issue by cyrillebenoit** [kbuzsaki/bingosync#127]
Hi,

I'm working on a browser extension for Bingosync and I need to retrieve the lists of goals for the current game and variant in order to add a couple new features (organizing the card, and custom translation notably)

As of right now, I've been able to fetch one of the generators, and then evaluate the code, but it's very janky and unsafe since I'm using eval on code pulled from the web.

On top of that, this method does not work from the extension options page, where the user would input the url of the generator if I had to go with the current system. (CSP protecting the browser)

![image](https://user-images.githubusercontent.com/11501201/101847606-e1b51f00-3b53-11eb-8e24-2f6989e46a78.png)

A way more elegant and practice way to do this would be to be able to read current Game and Variant somewhere in the DOM, and call an internal API that returns every list that the generator used.

Is this a thing ? Could you open an API for it if it isn't ?

Let me know if you need any more details about what I'm trying to accomplish.

**Comment by kbuzsaki** [comment]
Are you saying that you want the full list of goals for that generator, not just the goals on the current card?

Bingosync’s model for generators is that they are opaque javascript programs with a specified generator function signature. Most of the time those generators follow a convention where they declare their goal list as as variable named `bingoList`, but it's not a requirement, and some generators definitely have their goal list in a different formst captured in the function closure, with a dummy value passed in for `bingoList`.

**Comment by cyrillebenoit** [comment]
Yes that's what I wanted, I built around this limitation by fetching data at [these types of links](https://raw.githubusercontent.com/kbuzsaki/bingosync/master/bingosync-app/generators/super_mario_sunshine_1v1_generator.js) and returning the bingoList object.

Some issues with this method are :
1. I have to check the list's shape myself
2. The user has to input the URL for the generator and may make a mistake
3. If the user changes game / variant, they'll have to go back to the settings page for the extension and manually change to the new generator link.

An API would be able to solve these issues by:

1. Having a tag somewhere in the room DOM with the current Game and Variant.
2. Having a tag somewhere in the room DOM with other available Variants for that Game.
3. Given a pair {Game, Variant}, a call to the Lists API would return the bingoList and shape of it.

Again I build around so I guess we can close the issue, if you ever make this a thing, let me know so I can update my code to use it.

FYI, [here's a tour](https://youtu.be/lOdipr8t4ls) of features for the extension I made for Bingosync.
